backbone.js

```diff
@@ -1493,6 +1493,25 @@
   // Cached regex for stripping urls of hash.
   var pathStripper = /#.*$/;
 
+  // Cached regex for detecting any character that is invalid in an encoded URI.
+  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURI
+  var invalidURIEncoded = /[^;,/?:@&=+$a-zA-Z0-9-_.!~*'()#%]/;
+
+  // Cached regex for detecting invalid percent-encoded octets. Will match any
+  // "%" that is not followed by a hexadecimal number.
+  var invalidPercentOctets = /%(?![a-f0-9]{2}|[A-F0-9]{2})/;
```

Comment:
I'd like to get your thoughts on this. While `%Ab` is technically a valid percent-encoding, I imagine a respectable browser always using the same casing. In particular, requiring the same case could catch "encoded" strings like `1.5%Fat` (cause I can't think of a good example...).